### PR TITLE
ICU-20460 Adding mechanism to build unicore data into dat file.

### PR DESCRIPTION
**Pull request by sffc**
##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/ICU-20460
- [x] Updated PR title and link in previous line to include Issue number
- [x] Issue accepted
- [ ] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

File: icu4c/source/data/icu4j-readme.txt [hunk 1]
```diff
@@ -17,8 +17,14 @@ In the following,
         $icu4j_root is the ICU4J root directory
         $jdk_bin is the JDK bin directory (for the jar tool)
 
-1. Download and build ICU4C. For more instructions on downloading and building
-        ICU4C, see the ICU4C readme at:
+1. Download, configure, and build ICU4C. When you configure ICU4C, you must
+   set the environment variable ICU_DATA_BUILDTOOL_OPTS to
+   "--include_uni_core_data" to build additional required ICU4J data:
+
+        ICU_DATA_BUILDTOOL_OPTS=--include_uni_core_data ./runConfigureICU Linux
```
Comment (jefgen) [inline icu4c/source/data/icu4j-readme.txt]: Forgive my ignorance, but does this also work for platforms without the runConfigureICU script?

File: icu4c/source/data/BUILDRULES.py [hunk 2]
```diff
@@ -308,6 +310,36 @@ def generate_coll_ucadata(config, glob, common_vars):
     ]
 
 
+def generate_full_unicore_data(config, glob, common_vars):
+    # The core Unicode properties files (pnames.icu, uprops.icu, ucase.icu, ubidi.icu)
+    # are hardcoded in the common DLL and therefore not included in the data package any more.
+    # They are not built by default but need to be built for ICU4J data and for getting the .c source files
+    # when updating the Unicode data.
```
Comment (markusicu) [inline icu4c/source/data/BUILDRULES.py]: I believe that the part "and for getting the .c source files when updating the Unicode data" is no longer true.
Please remove.

Instead we can say something like we need these for ICU4J, both in the .jar and in the .dat file (if ICU4J uses the .dat file).

File: icu4c/source/data/BUILDRULES.py [hunk 3]
```diff
@@ -308,6 +310,36 @@ def generate_coll_ucadata(config, glob, common_vars):
...
+    # See Jitterbug 4497. (makedata.mak revision 1.117)
```
Comment (markusicu) [inline icu4c/source/data/BUILDRULES.py]: Jitterbug is what we used before Trac. Please refer to ICU-4497.
"revision 1.117" is from cvs, pre-svn ... I would just remove the parenthesis.

File: icu4c/source/data/buildtool/__main__.py [hunk 4]
```diff
@@ -122,6 +128,9 @@ def __init__(self, args):
         # Either "unihan" or "implicithan"
         self.coll_han_type = args.collation_ucadata
 
+        # Boolean: whether to include unicore data in dat file
```
Comment (markusicu) [inline icu4c/source/data/buildtool/__main__.py]: Whether to include core Unicode data files in the .dat file.

File: icu4c/source/data/icu4j-readme.txt [hunk 5]
```diff
@@ -17,8 +17,14 @@ In the following,
...
+
+   For more instructions on downloading and building ICU4C,
+   see the ICU4C readme at:
         http://source.icu-project.org/repos/icu/trunk/icu4c/readme.html#HowToBuild
```
Comment (markusicu) [inline icu4c/source/data/icu4j-readme.txt]: Please update this link to point to GitHub.

File: icu4c/source/data/icu4j-readme.txt (hunk 1 above)
Comment (sffc) [reply icu4c/source/data/icu4j-readme.txt]: I put code to pass the same `ICU_DATA_BUILDTOOL_OPTS` environment variable as command-line arguments to Python in both configure.ac and makedata.mak, but I did not test it on any platforms other than Debian Linux.

File: icu4c/source/data/icu4j-readme.txt (hunk 5 above)
Comment (sffc) [reply icu4c/source/data/icu4j-readme.txt]: Is htmlpreview.github.io ok?

File: icu4c/source/data/buildtool/__main__.py (hunk 4 above)
Comment (sffc) [reply icu4c/source/data/buildtool/__main__.py]: Done

File: icu4c/source/data/BUILDRULES.py (hunk 3 above)
Comment (sffc) [reply icu4c/source/data/BUILDRULES.py]: Done

File: icu4c/source/data/BUILDRULES.py (hunk 2 above)
Comment (sffc) [reply icu4c/source/data/BUILDRULES.py]: Done

**Comment by sffc** [comment]
Commit amended.

File: icu4c/source/data/icu4j-readme.txt (hunk 1 above)
Comment (jefgen) [reply icu4c/source/data/icu4j-readme.txt]: Thanks Shane. I don't have time to test this at the moment, but it feels like we can always just file a bug in the future if need be. thanks.

File: icu4c/source/data/icu4j-readme.txt (hunk 5 above)
Comment (markusicu) [reply icu4c/source/data/icu4j-readme.txt]: yes, tnx

**Pull request merged**